xlb/utils/mesher.py

```diff
@@ -346,9 +346,7 @@ def _merge_duplicates(self, coordinates, connectivity):
 
             # Simple hashing: grid coordinates as tuple keys
             grid_coords = np.round(coords_chunk / tolerance).astype(np.int64)
-            hash_keys = (grid_coords[:, 0] +
-                         grid_coords[:, 1] * 1_000_000 +
-                         grid_coords[:, 2] * 1_000_000_000_000)
+            hash_keys = grid_coords[:, 0] + grid_coords[:, 1] * 1_000_000 + grid_coords[:, 2] * 1_000_000_000_000
```

Comment:
To ensure an unique 1D mapping for any grid size, we can use the memory pitch of a point within the dense bounding box. 
Something like:
`key =  grid_coords[:, 0] + grid_coords[:, 1] * boxDim[0] + grid_coords[:, 2] * boxDim[0] * boxDim[1]
`